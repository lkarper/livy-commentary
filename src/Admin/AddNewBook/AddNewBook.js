import React, { useState, useContext } from 'react';
import CommentaryContext from '../../context/CommentaryContext';
import CommentaryService from '../../services/commentary-service';

const AddNewBook = (props) => {

    const context = useContext(CommentaryContext);

    const [newBookNumber, setNewBookNumber] = useState(props.nextBook);

    const handleSubmit = (event) => {
        event.preventDefault();
        CommentaryService.addNewBook(newBookNumber)
            .then(book => {
                context.addNewBook(book);
                props.setAddNewBook(false);
            })
            .catch(error => {
                console.log(error);
            });
    }

    return (
        <form
            onSubmit={handleSubmit}
        >
            <label htmlFor='new-book-number'>New book number: </label>
            <input
                type='number'
                id='new-book-number'
                value={newBookNumber}
                onChange={(e)  => setNewBookNumber(e.target.value)}
            />
            <button
                type='submit'
                disabled={!newBookNumber}
            >
                Submit
            </button>
        </form>
    )

}

export default AddNewBook;