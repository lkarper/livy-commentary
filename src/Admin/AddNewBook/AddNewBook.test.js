import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import AddNewBook from './AddNewBook';

describe('AddNewBook component', () => {
    it('renders the UI as expected', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <AddNewBook />,
            div
        );
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const wrapper = mount(<AddNewBook />);
        expect(toJSON(wrapper)).toMatchSnapshot();
        wrapper.unmount();
    });
});
