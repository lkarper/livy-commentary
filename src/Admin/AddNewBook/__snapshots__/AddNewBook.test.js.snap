// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddNewBook component renders the UI as expected 1`] = `
<AddNewBook
  nextBook={0}
  setAddNewBook={[Function]}
>
  <form
    onSubmit={[Function]}
  >
    <label
      htmlFor="new-book-number"
    >
      New book number: 
    </label>
    <input
      id="new-book-number"
      onChange={[Function]}
      type="number"
      value={0}
    />
    <button
      disabled={true}
      type="submit"
    >
      Submit
    </button>
  </form>
</AddNewBook>
`;
